### PR TITLE
Convert FastqToSam and RevertSam to use PicardHtsPath

### PR DESCRIPTION
**Pull request by markjschreiber**
### Description

Enables FastqToSam and RevertSam applications to use paths for input files.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/sam/RevertSam.java [hunk 1]
```diff
@@ -327,7 +328,11 @@ protected int doWork() {
         } else {
             final Map<SAMReadGroupRecord, FastqQualityFormat> readGroupToFormat;
             try {
-                readGroupToFormat = createReadGroupFormatMap(inHeader, REFERENCE_SEQUENCE, VALIDATION_STRINGENCY, INPUT, RESTORE_ORIGINAL_QUALITIES);
+                Path referenceSequencePath = null;
```
Comment (yfarjoun) [inline src/main/java/picard/sam/RevertSam.java:331]: make final and use `else` to set to null

File: src/test/java/picard/sam/AbstractAlignmentMergerTest.java [hunk 2]
```diff
@@ -697,7 +699,7 @@ public void testUnmapBacterialContamination() throws IOException {
 
         MergeBamAlignment mergeBamAlignment = new MergeBamAlignment();
 
-        mergeBamAlignment.ALIGNED_BAM = Collections.singletonList(file);
+        mergeBamAlignment.ALIGNED_BAM = Collections.singletonList(file.toPath().toFile()); // will update to use path when MergeBamAlignment is update to use Path
```
Comment (yfarjoun) [inline src/test/java/picard/sam/AbstractAlignmentMergerTest.java:702]: Make this a formal "TODO" (as a comment)

File: src/main/java/picard/sam/FastqToSam.java [hunk 3]
```diff
@@ -274,53 +280,53 @@ public static FastqQualityFormat determineQualityFormat(final FastqReader reader
...
         }
         if (null == fastqExtensions) {
-            throw new PicardException(String.format("Could not parse the FASTQ extension (expected '_001' + '%s'): %s", FastqExtensions.values().toString(), baseFastq));
+            throw new PicardException(String.format("Could not parse the FASTQ extension (expected '_001' + '%s'): %s", Arrays.toString(FastqExtensions.values()), baseFastq));
         }
 
         // Find all the files
         for (int idx = 2; true; idx++) {
-            String fastq = baseFastq.getAbsolutePath();
+            String fastq = baseFastq.toAbsolutePath().toString();
```
[the remark this answers is not in the document]
Comment (markjschreiber) [reply src/main/java/picard/sam/FastqToSam.java:303]: In this case yes because `baseFastq.toAbsolutePath()` will return a `Path` not a `String`

**Comment by markjschreiber** [comment]
Hi there, just wondering if this could be merged. I have some more conversions that I'll start working on soon.

**Comment by gbggrant** [comment]
Looks good to me - @yfarjoun do you want to re-review?

**Pull request merged**